mlir/lib/IR/BuiltinTypes.cpp

```diff
@@ -599,7 +599,7 @@ bool mlir::detail::isSupportedMemorySpace(Attribute memorySpace) {
     return true;
 
   // Supported built-in attributes.
-  if (memorySpace.isa<IntegerAttr, StringAttr, DictionaryAttr>())
+  if (memorySpace.isa<IntegerAttr, StringAttr, DictionaryAttr, FlatSymbolRefAttr>())
```

Comment:
I suggest using more generic `SymbolRefAttr` here. `FlatSymbolRefAttr` is a sub-class of it, so it will be covered too.